### PR TITLE
Support for GEO_LAT_LON instead of Z_ORDER_LAT_LON

### PR DESCRIPTION
**Pull request by geophile**
This is the sql-layer side of the slight generalization of support for functions in indexes. 

I have kept the enum, but added functionName to Index in ais.proto. The enum would work by itself except for getting case sensitivity correct. I'll remove it if anyone thinks this is pointless right now.

Tests run cleanly except as noted in recent email: some tests in fdb-sql-layer-pg are failing. Mike says these are known to occur occasionally, and they do seem far from my changes.

### CODE REVIEW

File: fdb-sql-layer-core/src/main/java/com/foundationdb/server/error/UnsupportedFunctionInIndexException.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (C) 2009-2013 FoundationDB, LLC
```
Comment (nathanlws) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/server/error/UnsupportedFunctionInIndexException.java]: 2015

**Comment by nathanlws** [comment]
> I'll remove it if anyone thinks this is pointless right now.

I do think it is fairly redundant with the enum. I don't think it would survive as-is when we do fully generalize otherwise I'd be fine leaving it. Normalizing the name to the enum (trim, case insensitive compare) at Table/IndexDDL time is enough for now I think.

Otherwise looks good.

File: fdb-sql-layer-core/src/main/java/com/foundationdb/server/error/UnsupportedFunctionInIndexException.java (hunk 1 above)
Comment (geophile) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/server/error/UnsupportedFunctionInIndexException.java]: I just copy/pasted another exception. Looks like many of these need to 
be updated:

jao@mintyzack ~/git/sql-layer $ find fdb-sql-layer-core/src -type f | 
grep "\.java" | xargs grep -l "2009-2013" | wc -l
1884

Comment (nathanlws) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/server/error/UnsupportedFunctionInIndexException.java]: This is a new file created this year.

The current edit is that if a file isn't modified its copyright doesn't need updating.

**Comment by nathanlws** [comment]
Thanks for the tweaks. Looks good now.

Approve

**Pull request merged**